### PR TITLE
Support TUF-only verification in aktualizr-secondary.

### PR DESCRIPTION
**Pull request by pattivacek**
There are plenty of test cases to verify this but I'm still open to feedback. ~~I also still need to work on the Primary side so that it doesn't try to send the Director metadata if it doesn't need to (and/or isn't available) in this case.~~

Update: Primary side works correctly now, too.

### CODE REVIEW

**Comment by pattivacek** [comment]
FYI @mike-sul @doanac. It also occurs to me that the version comparison logic might be similar to something you already have in aktualizr-lite.

File: src/aktualizr_secondary/aktualizr_secondary.cc [hunk 1]
```diff
@@ -65,23 +67,25 @@ data::InstallationResult AktualizrSecondary::install() {
...
-  // 3. NOT SUPPORTED: Download and check the Timestamp metadata file from the Director repository.
-  // 4. NOT SUPPORTED: Download and check the Snapshot metadata file from the Director repository.
-  // 5. Download and check the Targets metadata file from the Director repository.
-  try {
-    director_repo_.updateMeta(*storage_, metadata);
-  } catch (const std::exception& e) {
-    LOG_ERROR << "Failed to update Director metadata: " << e.what();
-    return data::InstallationResult(data::ResultCode::Numeric::kVerificationFailed,
-                                    std::string("Failed to update Director metadata: ") + e.what());
+  if (config_.uptane.verification_type == VerificationType::kFull) {
```
Comment (mike-sul) [inline src/aktualizr_secondary/aktualizr_secondary.cc:77]: I wonder if it makes sense to make it "Primary" driven, i.e. a decision whether to make a full or partial (just repo metadata) verification is made by Primary so it sends corresponding "instruction" to Secondary.

The  "instruction" could be
1. Additional parameter in the existing RPC call (AKIpUptaneMes_PR_putMetaReq*)
2. Presence or absence of directorRepo in AKPutMetaReq2Mes_t 
3. A new RPC call aimed specifically for delivering of just Repo metadata.

Comment (pattivacek) [reply src/aktualizr_secondary/aktualizr_secondary.cc:77]: I considered this, but the problem is that you introduce a security dependence on the Primary that we didn't have before. As in, if you were intending to do full verification, but an attacker manages to control the Primary and somehow force it to tell the Secondary to do something less the full verification, you've opened a hole that wasn't there before. The tradeoff with the way it is here is that you have a deeper dependence on the configuration and thus the filesystem, but that was already there from the beginning.

**Comment by pattivacek** [comment]
I can't figure out why now the first test case is looping indefinitely in of these complicated tests. Everything else looks fine.

File: include/libaktualizr/types.h [hunk 2]
```diff
@@ -457,6 +462,28 @@ class Manifest : public Json::Value {
   bool verifySignature(const PublicKey &pub_key) const;
 };
 
+static inline VerificationType VerificationTypeFromString(const std::string &vt_str) {
```
Comment (pattivacek) [inline include/libaktualizr/types.h:465]: These functions feel a bit dirty but I didn't think it was worth it to make a whole class just for these.

File: src/aktualizr_secondary/aktualizr_secondary_test.cc [hunk 3]
```diff
@@ -444,6 +454,71 @@ TEST_F(SecondaryTest, InvalidImageData) {
...
+  }
+}
+
+/**
+ * Instantiates the parameterized test for each specified value of
+ * std::pair<std::vector<std::string>, boost::optional<std::string>>>.
+ * The first parameter value is a list of Targets with custom versions and the
+ * second paramter is which one should be considered the latest, if any.
+ */
+INSTANTIATE_TEST_SUITE_P(SecondaryTestTufVersions, SecondaryTestTuf,
```
Comment (cajun-rat) [inline src/aktualizr_secondary/aktualizr_secondary_test.cc:498]: Cool!

**Pull request merged**